java/vector/src/main/java/org/apache/arrow/vector/BaseValueVector.java

```diff
@@ -98,5 +105,67 @@ protected ArrowBuf releaseBuffer(ArrowBuf buffer) {
   protected static int getValidityBufferSizeFromCount(final int valueCount) {
     return (int) Math.ceil(valueCount / 8.0);
   }
+
+  /* round up to the next multiple of 8 */
+  private static long roundUp8(long size) {
+    return ((size + 7) / 8) * 8;
+  }
+
+  protected long computeCombinedBufferSize(int valueCount, int typeWidth) {
```

Comment:
I'll add the pre-checks.

> Also, shouldn't we be returning the calculated valuecount? e.g. if you size for 50 but it turns > out we can fit 60 in the same memory, shouldn't we be using that as the available size/value > count (for initial alloc purposes, etc)?

That's done as part of allocDataAndValidityBufs (both for initial alloc and realloc) - in this case, it will return buffers with capacity of 60. And, the vector implementations will utilize the total capacity of 60.

The api setInitialCapacity() sets the variables based on whatever the user requests. Later, during alloc/realloc, the buffers allocated can have a capacity greater than this (power-of-2 roundup).